disnake/ext/commands/params.py

```diff
@@ -916,7 +1008,24 @@ def expand_params(command: AnySlashCommand) -> List[Option]:
         raise TypeError(f"Couldn't find an interaction parameter in {command.callback}")
 
     for injection in injections.values():
-        params += collect_nested_params(injection.function)
+        collected = collect_nested_params(injection.function)
+        if injection.autocompleters:  # != {}
+            for p in collected:
+                if f := injection.autocompleters.get(p.name):
+                    p.autocomplete = f
+
+            if len(injection.autocompleters) > len(collected):
+                raise ValueError(
+                    f"Couldn't set autocompleters on injection {injection.function.__qualname__} for non-existent"
+                    f" options: "
+                    + ", ".join(
+                        f"'{k}'"
+                        for k in _get_superfluous_autocompleters(
+                            collected, injection.autocompleters
+                        )
+                    )
+                )
```

Comment:
```suggestion
        if injection.autocompleters:
            lookup = {p.name: p for p in collected}
            for name, func in injection.autocompleters.items():
                param = lookup.get(name)
                if param is None:
                    raise ValueError(f"Option '{name}' doesn't exist in '{command.qualified_name}'")
                func.__slash_command__ = command
                param.autocomplete = func
```
Simplified it a bit, this avoids the need for something like `_get_superfluous_autocompleters`.
The addition of binding `__slash_command__` is important, since autocompletes would otherwise not work in cogs.